src/main/java/uk/ac/ic/wlgitbridge/server/Oauth2Filter.java

```diff
@@ -57,18 +69,22 @@ public void doFilter(
             ServletResponse servletResponse,
             FilterChain filterChain
     ) throws IOException, ServletException {
+        String requestUri = ((Request) servletRequest).getRequestURI();
+        if (requestUri.startsWith("/project")) {
```

Comment:
We discussed allowing the `/project` URL as well. It seems like that would be fairly easy to do here instead, and would probably be a smoother experience for the user. I know I'd find it useful to just be able to ctrl+c the URI from the address bar and paste into my console. (Currently, every time I clone a project I copy the git clone URL from the menu, type "git clone <paste>" and end up with "git clone git clone <url>")

We could say:

```java
if (requestUri.startsWith("/project")) {
        requestUri = requestUri.substring(8);
}

if (requestUri.length() - requestUri.replace("/", "").length() > 1) {
        // Invalid URI - contains more than one slash and does not start with `/project`
        Log.info("[{}] Invalid request URI", requestUri);
        sendResponse(servletResponse, 404, Arrays.asList(
                "Invalid Project URL. Please see your project page for the correct Git clone URL."
        ));
        return;
}

String project = Util.removeAllSuffixes(
        ((Request) servletRequest).getRequestURI().split("/")[1],
        requestUri.split("/")[1],
        ".git"
);

if (!project.matches("[0-9a-f]{12}")) {
        // Invalid URI - contains more than one slash and does not start with `/project`
        Log.info("[{}] Invalid project ID", requestUri);
        sendResponse(servletResponse, 404, Arrays.asList(
                "Invalid Project ID. Please see your project page for the correct Git clone URL."
        ));
        return;
}
```

Also, suggest 404 as a response as it's really an http request for something that doesn't exist, rather than a bad request.

WDYT?

Reply:
We discussed in the linked ticket, and remembered that we made this decision during the integration to adhere closer to the v1 api: https://github.com/overleaf/issues/issues/2031#issuecomment-516892840

Reply:
I guess 404 makes sense though :+1:

Reply:
Updated to use 404

Reply:
Have dropped in a comment in the linked issue to see if I could sway the argument a bit because I do prefer the supporting-multiple-urls thing.